tools/Azure.Mcp.Tools.AzureManagedLustre/tests/Azure.Mcp.Tools.AzureManagedLustre.UnitTests/FileSystem/FileSystemCreateCommandTests.cs

```diff
@@ -0,0 +1,362 @@
+    [InlineData("--subscription sub123 --resource-group rg1 --name amlfs-01 --location eastus --sku AMLFS-Durable-Premium-125 --subnet-id /subscriptions/sub123/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/vnet1/subnets/sub1 --zone 1 --maintenance-day Monday --maintenance-time 00:00", false)] // Missing size
+    [InlineData("--subscription sub123 --resource-group rg1 --name amlfs-01 --location eastus --sku AMLFS-Durable-Premium-125 --size 4 --zone 1 --maintenance-day Monday --maintenance-time 00:00", false)] // Missing subnet-id
+    [InlineData("--subscription sub123 --resource-group rg1 --name amlfs-01 --location eastus --sku AMLFS-Durable-Premium-125 --size 4 --subnet-id /subscriptions/sub123/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/vnet1/subnets/sub1 --maintenance-day Monday --maintenance-time 00:00", false)] // Missing zone
+    [InlineData("--subscription sub123 --resource-group rg1 --name amlfs-01 --location eastus --sku AMLFS-Durable-Premium-125 --size 4 --subnet-id /subscriptions/sub123/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/vnet1/subnets/sub1 --zone 1 --maintenance-time 00:00", false)] // Missing maintenance-day
+    [InlineData("--subscription sub123 --resource-group rg1 --name amlfs-01 --location eastus --sku AMLFS-Durable-Premium-125 --size 4 --subnet-id /subscriptions/sub123/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/vnet1/subnets/sub1 --zone 1 --maintenance-day Monday", false)] // Missing maintenance-time
+    public async Task ExecuteAsync_ValidatesInputCorrectly(string args, bool shouldSucceed)
+    {
+        // Arrange
+        if (shouldSucceed)
+        {
+            var expected = CreateLustre();
+            _svc.CreateFileSystemAsync(
+                Arg.Is(Sub), Arg.Is(Rg), Arg.Is(Name), Arg.Is(Location), Arg.Is(Sku), Arg.Is(Size), Arg.Is(SubnetId), Arg.Is(Zone),
+                Arg.Is("Monday"), Arg.Is("00:00"),
+                Arg.Any<string?>(), Arg.Any<string?>(), Arg.Any<string?>(),
+                Arg.Any<string?>(), Arg.Any<string?>(), Arg.Any<long?>(), Arg.Any<long?>(),
+                Arg.Any<bool>(), Arg.Any<string?>(), Arg.Any<string?>(), Arg.Any<string?>(),
+                Arg.Any<string?>(), Arg.Any<RetryPolicyOptions?>()).Returns(expected);
+        }
+
+        var parseResult = _commandDefinition.Parse(args.Split(' ', StringSplitOptions.RemoveEmptyEntries));
+
+        // Act
+        var response = await _command.ExecuteAsync(_context, parseResult);
+
+        // Assert
+        Assert.Equal(shouldSucceed ? 200 : 400, response.Status);
+        if (shouldSucceed)
+        {
+            Assert.NotNull(response.Results);
+            var json = JsonSerializer.Serialize(response.Results);
+            var result = JsonSerializer.Deserialize<CreateResultJson>(json);
```

Comment:
Let's not create custom test models for deserializing expected results, use JsonContext instead. This will provide validation of testing logic and serialization logic and be robust against any changes made to the model being returned.

```suggestion
            var result = JsonSerializer.Deserialize(json, AzureManagedLustreJsonContext.FileSystemCreateResult);
```

Reply:
Fixed